src/client/cli/cmd/restore.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "restore.h"
+#include "animated_spinner.h"
+#include "common_callbacks.h"
+#include "common_cli.h"
+
+#include <multipass/cli/argparser.h>
+#include <multipass/cli/prompters.h>
+#include <multipass/exceptions/cli_exceptions.h>
+
+#include <regex>
+
+namespace mp = multipass;
+namespace cmd = mp::cmd;
+
+namespace
+{
+const std::regex yes{"y|yes", std::regex::icase | std::regex::optimize};
+const std::regex no{"n|no", std::regex::icase | std::regex::optimize};
+} // namespace
+
+mp::ReturnCode cmd::Restore::run(mp::ArgParser* parser)
+{
+    if (auto ret = parse_args(parser); ret != ParseCode::Ok)
+        return parser->returnCodeFrom(ret);
+
+    AnimatedSpinner spinner{cout};
+
+    auto on_success = [this, &spinner](mp::RestoreReply& reply) {
+        spinner.stop();
+        fmt::print(cout, "Snapshot restored: {}.{}\n", request.instance(), request.snapshot());
+        return ReturnCode::Ok;
+    };
+
+    auto on_failure = [this, &spinner](grpc::Status& status) {
+        spinner.stop();
+        return standard_failure_handler_for(name(), cerr, status);
+    };
+
+    auto streaming_callback = [this,
+                               &spinner](mp::RestoreReply& reply,
```

Comment:
Hi, @townsend2010 
About this split, `BinPackArguments` parameter I have changed recently does have an impact on lambda capture list just like function arguments. However, it should not affect this case because `BinPackArguments` is about forbidding binpacking intermediate number arguments (the number of arguments is bigger than one and smaller than the total number of arguments) and this case only has two arguments. 

Now coming back to how to fix this. I am afraid this is no parameter that can make clang-format not to split capture list. Besides that, this case is kind of rare because the function parameter split did not do it then he splited capture list as well. I would say this current formatting mechanism on lambda is acceptable. If we want to fix this particular case, there are some tweaks we can do.

1. Add the trailing return type (which is also a good practice), which is `-> void` in this case and it can change the column length and change the formatting and avoid splitting the capture list.
2. Change the BraceWrapping: BeforeLambdaBody: to true
```
BraceWrapping:
  BeforeLambdaBody: true
```
This will always move the first `{` to the next line and change the formatting and avoid capture list split. This is an improvement to the lambda function formating in general because it gets consistent with our normal function, but this change will impact more files and we may not want to do it now. 
 3. Apply both of them. 

The item one is the quick fix and probably is what we want now.

Reply:
We could also add a using declaration to make that big parameter shorter in this case:

```cpp
using Client = grpc::ClientReaderWriterInterface<RestoreRequest, RestoreReply>;
auto streaming_callback = [this, &spinner](mp::RestoreReply& reply, Client* client) {
```

Reply:
Ok, let's do a quick fix either what @ricab proposes or @georgeliao's suggestion number 1.

I do think a separate PR should be submitted for suggestion 2 as I think I would prefer that formatting when it comes to lambdas.